yt_dlp/extractor/tvopengr.py

```diff
@@ -0,0 +1,150 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    get_elements_text_and_html_by_attribute,
+    unescapeHTML,
+)
+
+
+class TVOpenGrBaseIE(InfoExtractor):
+    def _return_canonical_url(self, url, video_id):
+        webpage = self._download_webpage(url, video_id)
+        canonical_url = self._og_search_url(webpage)
+        title = self._og_search_title(webpage)
+        self.to_screen(f'{video_id}: Redirect to canonical URL')
+        return self.url_result(canonical_url, ie=TVOpenGrWatchIE.ie_key(), video_id=video_id, video_title=title)
+
+
+class TVOpenGrWatchIE(TVOpenGrBaseIE):
+    IE_NAME = 'tvopengr:watch'
+    IE_DESC = 'tvopen.gr (and ethnos.gr) videos'
+    _VALID_URL = r'(?P<scheme>https?)://(?P<netloc>(?:www\.)?(?:tvopen|ethnos)\.gr)/watch/(?P<id>\d+)/(?P<slug>[^/]+)'
+    _API_ENDPOINT = '//www.tvopen.gr/templates/data/player'
+
+    _TESTS = [{
+        'url': 'https://www.ethnos.gr/watch/101009/nikoskaprabelosdenexoymekanenanasthenhsemethmethmetallaxhomikron',
+        'md5': '8728570e3a72e0f8d9475ba94859fdc1',
+        'info_dict': {
+            'id': '101009',
+            'title': 'md5:51f68773dcb6c70498cd326f45fefdf0',
+            'display_id': 'nikoskaprabelosdenexoymekanenanasthenhsemethmethmetallaxhomikron',
+            'description': 'md5:78fff49f18fb3effe41b070e5c7685d6',
+            'thumbnail': 'https://opentv-static.siliconweb.com/imgHandler/1920/d573ba71-ec5f-43c6-b4cb-d181f327d3a8.jpg',
+            'ext': 'mp4',
+            'upload_date': '20220109',
+            'timestamp': 1641686400,
+        },
+    }, {
+        'url': 'https://www.tvopen.gr/watch/100979/se28099agapaomenalla7cepeisodio267cmhthrargiapashskakias',
+        'md5': '38f98a1be0c577db4ea2d1b1c0770c48',
+        'info_dict': {
+            'id': '100979',
+            'title': 'md5:e021f3001e16088ee40fa79b20df305b',
+            'display_id': 'se28099agapaomenalla7cepeisodio267cmhthrargiapashskakias',
+            'description': 'md5:ba17db53954134eb8d625d199e2919fb',
+            'thumbnail': 'https://opentv-static.siliconweb.com/imgHandler/1920/9bb71cf1-21da-43a9-9d65-367950fde4e3.jpg',
+            'ext': 'mp4',
+            'upload_date': '20220108',
+            'timestamp': 1641600000,
+        },
+    }]
+
+    def _download_api_data(self, cid, scheme='https'):
+        url = f'{scheme}:{self._API_ENDPOINT}'
+        query = {'cid': cid}
+        return self._download_json(
+            url, query,
+            'Downloading JSON',
+            'Unable to download JSON',
+            query=query)
```

Comment:
```suggestion
        return self._download_json(url, query, query=query)
```
this note is basically same as the default

Also, `query` in place of video id?

Reply:
I also noticed this function is only ever used once. Since this is just a very thin wrapper, I dont think this is needed in the first place

Reply:
`query` was wrong, thanks.

I agree about thin wrapper function, I will remove it and call `_download_json` directly.